### DISCUSSION TITLE
v0.6.6

### DISCUSSION THREAD

**Discussion by takatost** [langgenius/dify#4053] [Releases]
## ✨ What’s New in v0.6.6? ✨

This update primarily focuses on enhancing system stability and improving user experience. Key updates include:

### 🚀 New Features

* Tool Firecrawl support by @richards199999 in #3819

  This new tool provides capabilities akin to JinaReader, allowing you to efficiently scrape web content and convert it to Markdown format.

* Bedrock Llama3 support by @longzhihun in #3890
* Minimax abab6.5, abab6.5s support by @Weaxs in #4012
* VectorDB PGVecto_rs support and SQLAlchemy upgrade to 2.x by @JohnJyong in #3833
* Storage gcs support by @miendinh in #3887
* Added support for HTTP Request Timeout Settings in the HTTP request node by @majian159 in #3854

  With the new connection, read, and write timeout settings, you can control your HTTP requests with precision.

### ⚙️ Enhancements
* Added support for Traditional Chinese language by @yongjer in #3899
* Workflow remove preview mode by @zxhlyh in #3941
  
  During preview, users can modify workflow graph configurations to enhance the user experience.”

* Qdrant gRPC mode support by @leslie2046 in #3929
* Refactor storage using the factory pattern by @JohnJyong in #3922
* Update sandbox service image tag to a specific version to avoid any issues with delayed image updates by @Yeuoly in #3997
* Text stream support for workflow app when using LLM node text as End node output by @takatost in #3798

  If the END node in the workflow app contains the LLM text variable, it can support streaming output in typewriter form.

* CODE transform node editor support insert var by add slash or left brace by @iamjoel in #3946
* Added class_name output variable for question classifier node by @sinomoe in #4000

### 🛠️ Bug Fixes

* fix: tool webscraper - too many redirects in case target url does not support HEAD method and redirect infinite to the same error link by @miendinh in #3831
* fix: fetch page name of notion wiki by @badbye in #3847
* fix: together ai model setting by @JohnJyong in #3895
* feat: replicate model provider supports default version by @GarfieldDai in #3884

For additional enhancements and updates, refer to the What's Changed section below.

### Docker compose deployments:

1. Get the latest code from the main branch:

   ```Shell
   git checkout main
   git pull origin main
   ```

2. Go to the next step and update to the latest image:

   ```Shell
   cd docker
   docker compose up -d
   ```

### Source Code deployments:

1. Stop API server, Worker and Web frontend Server.

2. Get the latest code from the main branch:

   ```Shell
   git checkout main
   git pull origin main
   ```

3. Update Python dependencies:

   ```Shell
   cd api
   pip install -r requirements.txt
   ```

4. Then, let's run the migration script:

   ```Shell
   flask db upgrade
   ```

5. Finally, run API server, Worker and Web frontend Server again.

## What's Changed
* test: refactor vdb tests by visitor design pattern by @bowenliang123 in https://github.com/langgenius/dify/pull/3838
* Reduce tool redundancy for [Judge0 CE] by @richards199999 in https://github.com/langgenius/dify/pull/3837
* fix: using api can not execute relyt vector database by @klaus-xiong in https://github.com/langgenius/dify/pull/3766
* fix: billing tenant account role. by @GarfieldDai in https://github.com/langgenius/dify/pull/3850
* refactor: config file by @takatost in https://github.com/langgenius/dify/pull/3852
* test: add tests covering all methods of vector store by @bowenliang123 in https://github.com/langgenius/dify/pull/3849
* fix: chat api doc not show title in english vision by @iamjoel in https://github.com/langgenius/dify/pull/3864
* fix: UP031 style rule violation by @bowenliang123 in https://github.com/langgenius/dify/pull/3866
* optimize the knowledge failed documents query by @charli117 in https://github.com/langgenius/dify/pull/3870
* fix: full screen editor not follow panel width by @iamjoel in https://github.com/langgenius/dify/pull/3876
* fix: copy invite link for HTTPS has deplicate origin by @zhangever in https://github.com/langgenius/dify/pull/3877
* add the comment in entrypoint.sh by @hikariming in https://github.com/langgenius/dify/pull/3882
* fix: tool webscraper - too many redirects in case target url does not… by @miendinh in https://github.com/langgenius/dify/pull/3831
* fix: fetch page name of notion wiki by @badbye in https://github.com/langgenius/dify/pull/3847
* test: improve vector store tests by @bowenliang123 in https://github.com/langgenius/dify/pull/3855
* add together ai model setting by @JohnJyong in https://github.com/langgenius/dify/pull/3895
* chore: integrate code-inspector-plugin by @TinsFox in https://github.com/langgenius/dify/pull/3900
* Add support for Traditional Chinese language by @yongjer in https://github.com/langgenius/dify/pull/3899
* feat: replicate supports default version. by @GarfieldDai in https://github.com/langgenius/dify/pull/3884
* fix: hydration warning by @TinsFox in https://github.com/langgenius/dify/pull/3897
* feat: add support for Bedrock LLAMA3 by @longzhihun in https://github.com/langgenius/dify/pull/3890
* chore: skip warning messages when pytest auto-collecting the vdb test class by removing `Test` prefix by @bowenliang123 in https://github.com/langgenius/dify/pull/3906
* feat: Add google storage support by @miendinh in https://github.com/langgenius/dify/pull/3887
* fix: api doc example error by @StudyingLover in https://github.com/langgenius/dify/pull/3925
* fix: code full screen in web app cause error by @iamjoel in https://github.com/langgenius/dify/pull/3935
* feat: add start commands to devcontainer by @perzeuss in https://github.com/langgenius/dify/pull/3902
* improvement: Optimizing the experience of the app list page by @majian159 in https://github.com/langgenius/dify/pull/3885
* enhancement:support  Qdrant gRPC mode by @leslie2046 in https://github.com/langgenius/dify/pull/3929
* feat: workflow remove preview mode by @zxhlyh in https://github.com/langgenius/dify/pull/3941
* feat: llm text stream support for workflow app by @takatost in https://github.com/langgenius/dify/pull/3798
* optimize: code node has a bad error message by @Yeuoly in https://github.com/langgenius/dify/pull/3949
* feat: code transform node editor support insert var by add slash or left brace by @iamjoel in https://github.com/langgenius/dify/pull/3946
* chore: update `@types/react` by @TinsFox in https://github.com/langgenius/dify/pull/3939
* feat: rename var name sync to used jinjia code by @iamjoel in https://github.com/langgenius/dify/pull/3964
* delete duplicate check get_dataset by @baxiang in https://github.com/langgenius/dify/pull/3966
* add pgvecto_rs support and upgrade SQLAlchemy by @JohnJyong in https://github.com/langgenius/dify/pull/3833
* fix: db connections not being released during workflow execution by @takatost in https://github.com/langgenius/dify/pull/3971
* fix-nvidia-llama3 by @joshua20231026 in https://github.com/langgenius/dify/pull/3973
* feat: add support for request timeout settings in the HTTP request node. by @majian159 in https://github.com/langgenius/dify/pull/3854
* Add new tool: Firecrawl by @richards199999 in https://github.com/langgenius/dify/pull/3819
* Fix: password check in page of install by @JzoNgKVO in https://github.com/langgenius/dify/pull/3978
* add default value for QDRANT_GRPC_PORT by @JohnJyong in https://github.com/langgenius/dify/pull/3976
* Feat/install process refinement by @crazywoola in https://github.com/langgenius/dify/pull/3982
* add storage factory by @JohnJyong in https://github.com/langgenius/dify/pull/3922
* fix: ci by @crazywoola in https://github.com/langgenius/dify/pull/3983
* chore: add sandbox tag by @Yeuoly in https://github.com/langgenius/dify/pull/3997
* fix: workflow disable shortcuts when feature panel occured by @zxhlyh in https://github.com/langgenius/dify/pull/4001
* fix: prompt editor insert quickly by @zxhlyh in https://github.com/langgenius/dify/pull/4004
* Enabled Notion integration setup in Docker Compose Deployment by @kei-yamazaki in https://github.com/langgenius/dify/pull/3919
* fix(web): copy button visible at chat page normally by @JunIce in https://github.com/langgenius/dify/pull/4005
* Update the description and labels in Judge0ce tool by @richards199999 in https://github.com/langgenius/dify/pull/3990
* fix: submitCodeExecutionTask by @crazywoola in https://github.com/langgenius/dify/pull/4006
* chore: skip deprecated field_schema param in creating payload index on Qdrant by @bowenliang123 in https://github.com/langgenius/dify/pull/3903
* fix: prompt undefined caused match problem by @iamjoel in https://github.com/langgenius/dify/pull/4010
* feat: support minimax abab6.5, abab6.5s by @Weaxs in https://github.com/langgenius/dify/pull/4012
* feat: support question classifier node output by @sinomoe in https://github.com/langgenius/dify/pull/4000
* add glm-3-turbo max_tokens parameter setting by @orangeclk in https://github.com/langgenius/dify/pull/4017
* chore(deps): bump word-wrap from 1.2.3 to 1.2.5 in /web by @dependabot in https://github.com/langgenius/dify/pull/4018
* fix: lodash version has warning by @iamjoel in https://github.com/langgenius/dify/pull/4020
* fix: typo of jinja2 by @bowenliang123 in https://github.com/langgenius/dify/pull/4019
* chore(deps): bump semver from 5.7.1 to 5.7.2 in /web by @dependabot in https://github.com/langgenius/dify/pull/4022
* test: remove explicit env settings for CI pytests by @bowenliang123 in https://github.com/langgenius/dify/pull/4041
* version to 0.6.6 by @takatost in https://github.com/langgenius/dify/pull/4050

## New Contributors
* @zhangever made their first contribution in https://github.com/langgenius/dify/pull/3877
* @hikariming made their first contribution in https://github.com/langgenius/dify/pull/3882
* @badbye made their first contribution in https://github.com/langgenius/dify/pull/3847
* @TinsFox made their first contribution in https://github.com/langgenius/dify/pull/3900
* @yongjer made their first contribution in https://github.com/langgenius/dify/pull/3899
* @StudyingLover made their first contribution in https://github.com/langgenius/dify/pull/3925
* @majian159 made their first contribution in https://github.com/langgenius/dify/pull/3885
* @baxiang made their first contribution in https://github.com/langgenius/dify/pull/3966
* @kei-yamazaki made their first contribution in https://github.com/langgenius/dify/pull/3919

**Full Changelog**: https://github.com/langgenius/dify/compare/0.6.5...0.6.6

<hr /><em>This discussion was created from the release <a href='https://github.com/langgenius/dify/releases/tag/0.6.6'>v0.6.6</a>.</em>

**Comment by hanfangyuan4396** [comment]
👍🏻
